pkg/sdk/v1beta1/kafkacluster_types.go

```diff
@@ -1061,7 +1066,9 @@ func (mConfig *MonitoringConfig) GetCCJMXExporterConfig() string {
 }
 
 // GetBrokerConfig composes the brokerConfig for a given broker using the broker's config group
-func (b *Broker) GetBrokerConfig(brokerConfigGroups map[string]BrokerConfig) (*BrokerConfig, error) {
+func (b *Broker) GetBrokerConfig(kafkaClusterSpec KafkaClusterSpec) (*BrokerConfig, error) {
```

Comment:
Instead of passing in the entire `KafkaClusterSpec` why don't we pass in only what's needed by this method? e.g. `GetBrokerConfig(brokerConfigGroups map[string]BrokerConfig, envs []corev1.EnvVar)`

Reply:
I thought about this too, but I have the impression that we may need further fields from `KafkaCluster` in the future. Also note, that if we do this later, it will brake the API again when we change that. In my opinion it is safer to leave it like this - so we can adjust the implementation later without breaking the syntax if we need some additional behaviour.